src/main/java/roomescape/dao/ReservationDao.java

```diff
@@ -0,0 +1,73 @@
+package roomescape.dao;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+import roomescape.domain.Reservation;
+import roomescape.domain.Time;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+
+@Repository
+public class ReservationDao {
+    private final JdbcTemplate jdbcTemplate;
+
+    public ReservationDao(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    private final RowMapper<Reservation> reservationRowMapper = (resultSet, rowNum) -> {
+        Time time = new Time(
+                resultSet.getLong("time_id"),
+                resultSet.getString("time_value")
+        );
+        return new Reservation(
+                resultSet.getLong("id"),
+                resultSet.getString("name"),
+                resultSet.getString("date"),
+                time
+        );
+    };
+
+    public List<Reservation> findAll() {
+        String sql = """
+            SELECT r.id, r.name, r.date, t.id as time_id, t.time as time_value
+            FROM reservation r
+            INNER JOIN time t ON r.time_id = t.id
+        """;
+        return jdbcTemplate.query(sql, reservationRowMapper);
+    }
+
+    public Reservation findById(Long id) {
+        String sql = """
+            SELECT r.id, r.name, r.date, t.id as time_id, t.time as time_value
+            FROM reservation r
+            INNER JOIN time t ON r.time_id = t.id
+            WHERE r.id = ?
+        """;
+        return jdbcTemplate.queryForObject(sql, reservationRowMapper, id);
+    }
+
+    public void delete(Long id) {
+        String sql = "DELETE FROM reservation WHERE id = ?";
+        jdbcTemplate.update(sql, id);
+    }
```

Comment:
```suggestion
    public void deleteById(Long id) {
        String sql = "DELETE FROM reservation WHERE id = ?";
        jdbcTemplate.update(sql, id);
    }
```

메서드명을 좀 더 명확히 작성하면 좋을 것 같아요! 

문자열 블럭을 활용하여 sql을 작성하신 것은 좋은 것 같아요!